xmppserver/src/main/java/org/jivesoftware/openfire/server/ServerDialback.java

```diff
@@ -680,6 +680,9 @@ private VerifyResult sendVerifyKey(String key, StreamID streamID, String recipie
                 doc = reader.parseDocument();
             } catch (DocumentException e) {
                 log.warn("Unable to verify key: XML Error!", e);
+                // Close the stream
+                writer.write("</stream:stream>");
+                writer.flush();
```

Comment:
```
writer.write("</stream:stream>");
writer.flush();
```
Appears a number of times in this file. I wonder if it's worth creating a utility method, `closeStreamAndFlush(writer)` so this is all captured in one place?

Reply:
That's not reducing complexity, that's just moving complexity around :)